### ISSUE TITLE
aws_launch_template does not apply security groups defined in network_interfaces block

### ISSUE THREAD

**Issue by george-richardson** [terraform-providers/terraform-provider-aws#4358]
### Terraform Version
Terraform v0.11.0
+ provider.aws v1.16.0

### Affected Resource(s)

- aws_launch_template

### Terraform Configuration Files
```hcl
resource "aws_launch_template" "example_failing_launch_template" {
  name          = "example_failing_launch_template"
  image_id      = "ami-5718412e"
  instance_type = "t2.micro"
  key_name      = "some_key_name"

  network_interfaces {
    associate_public_ip_address = true
    subnet_id                   = "subnet-05afea5e"
    security_groups             = ["sg-6f372715"]
    delete_on_termination       = true
  }
}

```

### Debug Output
https://gist.github.com/george-richardson/9b9d7f7701433c8fd04c465c88b851f7

### Expected Behavior
Terraform creates an `aws_launch_template` which has an associated network interface with the security group "sg-6f372715".

### Actual Behavior
The launch template is created, as is the network interface. However the security group information is missing. Here is a screenshot from the AWS console showing the network interfaces section of the generated launch template.

![image](https://user-images.githubusercontent.com/5854391/39300422-8925f300-4943-11e8-8f82-0e82c8858b00.png)

**Comment by bflad** [comment]
The fix for this has been merged into master and will release with v1.17.0 of the AWS provider, likely mid next week.

**Comment by george-richardson** [comment]
Thanks @bflad and @kl4w for such a quick turn around 😄

**Comment by bflad** [comment]
This has been released in [version 1.17.0 of the AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#1170-may-02-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
